src/util/format-ref.util.ts

```diff
@@ -1,18 +1,51 @@
 import { Reference } from "../model/reference.model";
 import { getDate } from "./get-date.util";
+import { escapeTitleText, escapeLinkUrl } from "./markdown.util";
 
 /**
  * Format a given reference as a valid markdown url
  */
 export function formatReference (ref: Reference): string {
-    const author: string = ref.author[0].family;
-    const year: number = getDate(ref).getFullYear();
-    const title: string = ref.title.split(" ").join("+");
+    const displayText: string = escapeTitleText(getDisplayText(ref));
+    const link: string = escapeLinkUrl(getLink(ref));
+    return `[@${displayText}](${link})`;
+}
+
+/**
+ * Given a reference, returns the family name of the first author
+ * If there are no authors provided, returns empty string ("")
+ */
+function getDisplayText (ref: Reference): string {
+    const author = getAuthor(ref);
+    let text: string = author;
+    try {
+        text += getDate(ref).getFullYear();
+    } catch (e) { }
+
+    // If there is no author or year, put the title of the ref
+    if (author === "") text = ref.title.split(" ").join();
+    return text;
+}
 
+/**
+ * returns the url of the ref to be used in markdown
+ */
+function getLink (ref: Reference): string {
     let url: string = "";
-    if (ref.DOI && ref.DOI !== "") url = `http://dx.doi.org/${ref.DOI}`;
-    else if (ref.URL && ref.URL !== "") url = ref.URL;
-    else url = `https://scholar.google.com/scholar?q=${title},+${author}`;
-    
-    return `[@${author + year}](${url})`;
+    if (ref.DOI && ref.DOI !== "") {
+        url = `http://dx.doi.org/${ encodeURI(ref.DOI) }`;
```

Comment:
Please check the difference between encodeURI and encodeURIComponent: https://love2dev.com/blog/whats-the-difference-between-encodeuri-and-encodeuricomponent/

In this case it should be encodeURIComponent because it's a query parameter.